File: qz/u2/santi/Exam U2/71-100/ExamSpeakers/src/main/java/ec/edu/espe/examspeakers/view/MainMenuView.java

package ec.edu.espe.examspeakers.view;

import ec.edu.espe.examspeakers.controller.SpeakerController;
import java.awt.BorderLayout;
import java.awt.FlowLayout;
import java.awt.Font;
import javax.swing.BorderFactory;
import javax.swing.JFrame;
import javax.swing.JLabel;
import javax.swing.JMenu;
import javax.swing.JMenuBar;
import javax.swing.JMenuItem;
import javax.swing.JOptionPane;
import javax.swing.JPanel;
import javax.swing.SwingConstants;
import javax.swing.UIManager;

/**
 *
 * @author Sjean
 */
public class MainMenuView extends JFrame {
    private SpeakerController controller;

    public MainMenuView(SpeakerController controller) {
        this.controller = controller;
        
        setTitle("Speaker Management System");
        setSize(800, 600);
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        
        try {
            for (UIManager.LookAndFeelInfo info : UIManager.getInstalledLookAndFeels()) {
                if ("Nimbus".equals(info.getName())) {
                    UIManager.setLookAndFeel(info.getClassName());
                    break;
                }
            }
        } catch (Exception e) {
            try {
                UIManager.setLookAndFeel(UIManager.getCrossPlatformLookAndFeelClassName());
            } catch (Exception ex) {
                
            }
        }

        JMenuBar menuBar = new JMenuBar();
        setJMenuBar(menuBar);
        JMenu fileMenu = new JMenu("File");
        JMenu speakerMenu = new JMenu("Speakers");
        JMenu helpMenu = new JMenu("Help");
        menuBar.add(fileMenu);
        menuBar.add(speakerMenu);
        menuBar.add(helpMenu);
        
        JMenuItem exitItem = new JMenuItem("Exit");
        JMenuItem createSpeakerItem = new JMenuItem("Create Speaker");
        JMenuItem listSpeakersItem = new JMenuItem("List Speakers");
        JMenuItem aboutItem = new JMenuItem("About");
        fileMenu.add(exitItem);
        speakerMenu.add(createSpeakerItem);
        speakerMenu.add(listSpeakersItem);
        helpMenu.add(aboutItem);
    
        exitItem.addActionListener(e -> System.exit(0));
        createSpeakerItem.addActionListener(e -> new CreateSpeakerView(controller).setVisible(true));
        listSpeakersItem.addActionListener(e -> new ListSpeakersView(controller).setVisible(true));
        aboutItem.addActionListener(e -> JOptionPane.showMessageDialog(this, "Speaker Management System\nVersion 1.0\nDeveloped by Your Name", "About", JOptionPane.INFORMATION_MESSAGE));
      
        JPanel mainPanel = new JPanel(new BorderLayout());
        setContentPane(mainPanel);
    
        JLabel welcomeLabel = new JLabel("Welcome to Speaker Management System", SwingConstants.CENTER);
        welcomeLabel.setFont(new Font("Arial", Font.BOLD, 24));
        mainPanel.add(welcomeLabel, BorderLayout.CENTER);
    
        JPanel statusBar = new JPanel(new FlowLayout(FlowLayout.LEFT));
        statusBar.setBorder(BorderFactory.createLoweredBevelBorder());
        JLabel statusLabel = new JLabel("Ready");
        statusBar.add(statusLabel);
        mainPanel.add(statusBar, BorderLayout.SOUTH);
    }
}

File: qz/u3/sigcha/qz1&2/Quizz_Speakers/Quizz_Speakers/view/main_menu_view.py
import tkinter as tk
from tkinter import messagebox
from controller.speaker_controller import SpeakerController
from view.create_speaker_view import CreateSpeakerView
from view.list_speakers_view import ListSpeakersView

class MainMenuView(tk.Tk):
    def __init__(self, controller: SpeakerController):
        super().__init__()
        self.controller = controller
        
        self.title("Speaker Management System")
        self.geometry("800x600")
        
        self.create_widgets()

    def create_widgets(self):
        menu_bar = tk.Menu(self)
        self.config(menu=menu_bar)

        file_menu = tk.Menu(menu_bar, tearoff=0)
        speaker_menu = tk.Menu(menu_bar, tearoff=0)
        help_menu = tk.Menu(menu_bar, tearoff=0)

        menu_bar.add_cascade(label="File", menu=file_menu)
        menu_bar.add_cascade(label="Speakers", menu=speaker_menu)
        menu_bar.add_cascade(label="Help", menu=help_menu)

        file_menu.add_command(label="Exit", command=self.quit)
        speaker_menu.add_command(label="Create Speaker", command=self.open_create_speaker_view)
        speaker_menu.add_command(label="List Speakers", command=self.open_list_speakers_view)
        help_menu.add_command(label="About", command=self.show_about_info)

        main_panel = tk.Frame(self)
        main_panel.pack(fill=tk.BOTH, expand=True)

        welcome_label = tk.Label(main_panel, text="Welcome to Speaker Management System", font=("Arial", 24), pady=20)
        welcome_label.pack(pady=50)

        status_bar = tk.Frame(self, bd=1, relief=tk.SUNKEN)
        status_bar.pack(side=tk.BOTTOM, fill=tk.X)
        status_label = tk.Label(status_bar, text="Ready", anchor=tk.W)
        status_label.pack(side=tk.LEFT, padx=10)

    def open_create_speaker_view(self):
        CreateSpeakerView(self.controller).mainloop()

    def open_list_speakers_view(self):
        ListSpeakersView(self.controller).mainloop()

    def show_about_info(self):
        messagebox.showinfo("About", "Speaker Management System\nVersion 1.0\nDeveloped by Your Name")
